diff --git a/rootdir/etc/init.qcom-common.rc b/rootdir/etc/init.qcom-common.rc
index 5b584b9..d3ad4d7 100644
--- a/rootdir/etc/init.qcom-common.rc
+++ b/rootdir/etc/init.qcom-common.rc
@@ -383,9 +383,9 @@ on property:service.bootanim.exit=1
 on property:init.svc.bootanim=stopped
     # Set light thermal restrictions while optimizing apps
     # [throttle_freq, throttle_temp, unthrottle_temp]
-    write /sys/kernel/msm_thermal/low_thresh "1728000 45 43"
-    write /sys/kernel/msm_thermal/mid_thresh "1574400 50 47"
-    write /sys/kernel/msm_thermal/high_thresh "1267200 55 53"
+    write /sys/kernel/msm_thermal/zone0 "1728000 45 43"
+    write /sys/kernel/msm_thermal/zone1 "1574400 50 47"
+    write /sys/kernel/msm_thermal/zone2 "1267200 55 53"
     write /sys/kernel/msm_thermal/enabled 1
 
 on property:sys.boot_completed=1
@@ -410,10 +410,15 @@ on property:sys.boot_completed=1
     write /dev/cpuctl/cpu.notify_on_migrate 1
     write /sys/block/mmcblk0/queue/scheduler deadline
     write /sys/kernel/msm_thermal/enabled 0
-    write /sys/kernel/msm_thermal/low_thresh "1574400 40 37"
-    write /sys/kernel/msm_thermal/mid_thresh "1267200 46 43"
-    write /sys/kernel/msm_thermal/high_thresh "1190400 52 48"
-    write /sys/kernel/msm_thermal/sampling_ms 10000
+    write /sys/kernel/msm_thermal/zone0 "1728000 39 38"
+    write /sys/kernel/msm_thermal/zone1 "1574400 42 40"
+    write /sys/kernel/msm_thermal/zone2 "1497600 45 43"
+    write /sys/kernel/msm_thermal/zone3 "1267200 49 46"
+    write /sys/kernel/msm_thermal/zone4 "1190400 53 50"
+    write /sys/kernel/msm_thermal/zone5 "1036800 56 54"
+    write /sys/kernel/msm_thermal/zone6 "960000 62 60"
+    write /sys/kernel/msm_thermal/zone7 "729600 79 75"
+    write /sys/kernel/msm_thermal/sampling_ms 8000
     write /sys/kernel/msm_thermal/enabled 1
     start mpdecision
 
