diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index f4620c2..dab0b7b 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -1787,7 +1787,7 @@ public class VideoModule implements CameraModule,
             RotateTextToast.makeText(mActivity, R.string.error_app_unsupported_video_hdr,
                     Toast.LENGTH_LONG).show();
             mStartRecPending = false;
-            return;
+            return false;
         }
 
         //??
